src/main/webapp/app/overview/course-archive/course-archive.component.ts

```diff
@@ -0,0 +1,124 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { Course } from 'app/entities/course.model';
+import { CourseManagementService } from '../../course/manage/course-management.service';
+import { HttpErrorResponse, HttpResponse } from '@angular/common/http';
+import { AlertService } from 'app/core/util/alert.service';
+import { onError } from 'app/shared/util/global.utils';
+import { Subscription } from 'rxjs';
+import { faAngleDown, faAngleUp, faArrowDownAZ, faArrowUpAZ, faQuestionCircle } from '@fortawesome/free-solid-svg-icons';
+import { sortCourses } from 'app/shared/util/course.util';
+import { SizeProp } from '@fortawesome/fontawesome-svg-core';
+import { ArtemisSharedModule } from 'app/shared/shared.module';
+import { ArtemisSharedComponentModule } from 'app/shared/components/shared-component.module';
+import { CourseCardHeaderComponent } from '../course-card-header/course-card-header.component';
+
+@Component({
+    selector: 'jhi-course-archive',
+    templateUrl: './course-archive.component.html',
+    styleUrls: ['./course-archive.component.scss'],
+    standalone: true,
+    imports: [ArtemisSharedModule, ArtemisSharedComponentModule, CourseCardHeaderComponent],
+})
+export class CourseArchiveComponent implements OnInit, OnDestroy {
+    private archiveCourseSubscription: Subscription;
+
+    courses: Course[];
+    semesters: string[];
+    fullFormOfSemesterStrings: { [key: string]: string } = {};
+    semesterCollapsed: { [key: string]: boolean } = {};
+    coursesBySemester: { [key: string]: Course[] } = {};
+    searchCourseText = '';
+    isSortAscending = true;
+    iconSize: SizeProp = 'lg';
+
+    //Icons
+    readonly faAngleDown = faAngleDown;
+    readonly faAngleUp = faAngleUp;
+    readonly faArrowDownAZ = faArrowDownAZ;
+    readonly faArrowUpAZ = faArrowUpAZ;
+    readonly faQuestionCircle = faQuestionCircle;
+
+    constructor(
+        private courseService: CourseManagementService,
+        private alertService: AlertService,
+    ) {}
```

Comment:
Also, here you could use the new features from the new client guidelines. So instead of the constructor, you could use e.g. 
`    private readonly courseService = inject(CourseManagementService);`
You also might be able to get rid of the subscription if you use signals